src/controllers/gatewayController.ts

```diff
@@ -0,0 +1,49 @@
+import {Router, Request, Response} from 'express';
+import HttpStatus from 'http-status-codes';
+import {GatewayMessageSchema} from "../validation/gateway-message.schema";
+import sendNotification from "../services/sendNotification";
+import {default as SubscriptionModel, SubscriptionModelType} from "../models/SubscriptionModel";
+import logger from "../utils/logger";
+
+const router: Router = Router();
+
+router.post('/', async (request: Request, response: Response) => {
+    const message = request.body;
+    const validation = GatewayMessageSchema.validate(message);
+
+    if (validation.error) {
+        return sendResponse(response, false, validation.error.toString(), HttpStatus.BAD_REQUEST);
+    }
+
+    const subscriptions: SubscriptionModelType[] = await SubscriptionModel
+        .find({
+            address: message.recipientAddress
+        }) as SubscriptionModelType[];
+
+    if (subscriptions.length === 0) {
+        return sendResponse(response, false, 'No subscriptions found.');
+    }
+
+    try {
+        for (let subscription of subscriptions) {
+            const data = JSON.parse(subscription.body);
+            await sendNotification(data, message.params);
```

Comment:
Do we really need to wait for each notification before return? Would be better to use promises and `Promise.all`? (parallel processing)